CHANGELOG.md

```diff
@@ -26,8 +26,8 @@ A full list of changes can be found below.
 A number of improvements in the usage of process functions, i.e., `calcfunction` and `workfunction`, have been added.
 Each subsection title is a link to the documentation for more details.
 
-#### [Variable arguments](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/processes/functions.html#variable-and-keyword-arguments)
-Variable arguments can be used in case the function should accept a list of inputs of unknown length.
+#### [Variadic arguments](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/processes/functions.html#variadic-and-keyword-arguments)
```

Comment:
The link actually points to the documentation and there it still uses `variable`. Could you please also update it there. The anchor wouldn't work anyway otherwise

Reply:
Cheers, I didn't notice the link was to the docs (I thought it was a self-reference).

I've adapted the docs now, but reading through them I get the feeling that by supporting variadic positional arguments we may now be able to remove some of it?

In my view, the AiiDA docs don't need to explain how variadic arguments work in Python, except for cases where process functions don't work as expected for regular functions (and you have removed some of this).

Let me know, and I can add another commit that removes some of the docs on the topic.

Reply:
> In my view, the AiiDA docs don't need to explain how variadic arguments work in Python, except for cases where process functions don't work as expected for regular functions (and they now work more "as expected").

I think it is fine to provide some background, especially in the topics section. Not all users are very Python proficient that they will be familiar with these concepts, and they do behave differently in process functions (for example how the link labels are determined). So I think it is justified to keep it for now.